[gearswap] Update export.lua

lili-ffxi opened the pull request:
Removed some code duplication and reordered some other code.

Added a new `copy`/`clipboard` flag, which makes GS export the table to clipboard instead of to a file.

INPUT NEEDED:
I also added two new flags, `noaugments` and `onlyaugmented` which respectively do:
- `noaugments`/`noaugs` exports the set without recording the augments on each item. Useful if you mostly have single-path items so you don't get a spam of `augments={'Path: A'}` and such; or if you already have augmented items saved.
- `onlyaugmented`/`onlyaugs` *only* exports items that have augments and ignores entirely non-augmented items. Useful if you need to make variables for your augmented gear, or in combination with the `wearable` flag to export all augmented items that your current job can equip, to make a new gearswap file, etc.

The issue I have is that I would like gearswap to print out a notice when provided one of these flags, but I can't come up with good ones myself 😄 input appreciated

cairthenn committed:
Add files via upload

RubenatorX committed:
revert breaking [config] changes

lili-ffxi committed:
Update export.lua

Removed some code duplication and reordered some other code.

Added a new `copy`/`clipboard` flag, which makes GS export the table to clipboard instead of to a file.

INPUT NEEDED:
I also added two new flags, `noaugments` and `onlyaugmented` which respectively do:
- `noaugments`/`noaugs` exports the set without recording the augments on each item (useful if you mostly have single-path items so you don't get a spam of `Path: A` and such; or if you already have augmented items saved)
- `onlyaugmented`/`onlyaugs` *only* exports items that have augments and ignores entirely non-augmented items (useful if you need to make variables for your augmented gear, or in combination with the `wearable` flag to export all augmented items that your current job can equip, to make a new gearswap file, etc)

The issue I have is that I would like gearswap to print out a notice when provided one of these flags, but I can't come up with good ones myself 😄 input appreciated

z16 commented on addons/GearSwap/export.lua:
```diff
@@ -24,9 +24,11 @@
     local item_list = T{}
-    local targinv,all_items,minify,all_sets,use_job_in_filename,use_subjob_in_filename,overwrite_existing,named_file
+    local targinv,all_items,wearable,all_sets,noaugments,onlyaugmented,minify,clipboard
+    local use_job_in_filename,use_subjob_in_filename,overwrite_existing,named_file
```
I know this is not your doing, but I would put these all on separate lines and initialize them with default values (`false` for most of them).

z16 commented on addons/GearSwap/export.lua:
```diff
@@ -24,9 +24,11 @@
+    local use_job_in_filename,use_subjob_in_filename,overwrite_existing,named_file
     if #options > 0 then
         for _,v in ipairs(options) do
             if S{'inventory','inv','i'}:contains(v:lower()) then
```
`v` is lowercased all the time. Define it once at the top and use that:
```lua
local option = v:lower()
```

lili-ffxi committed:
Update export.lua

Removed stray `log()`
Tweaked argument parsing.
Finalized displayed message.

lili-ffxi commented:
Still working on this, hopefully I can submit the new code soonTM.

lili-ffxi committed:
Update README.md

Fixed markdown
Added the newer options for the `export` command.

lili-ffxi committed:
Update export.lua

Sept 2025 update
cleans up code
adds the ability to export to a BG-Wiki Template format (for ease of pasting into bg-wiki guides)

lili-ffxi commented:
PR closed because of too many conflicts, reopened in #2469
